### PR TITLE
Docs: Clarify stylistic rule update policy

### PR DESCRIPTION
**Pull request by btmills**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

I clarified our policy on stylistic rule updates based on eslint/tsc-meetings#237.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -187,6 +187,15 @@ ESLint follows [semantic versioning](https://semver.org). However, due to the na
 
 According to our policy, any minor update may report more linting errors than the previous release (ex: from a bug fix). As such, we recommend using the tilde (`~`) in `package.json` e.g. `"eslint": "~3.1.0"` to guarantee the results of your builds.
 
+## <a name="stylistic-rule-updates"></a>Stylistic Rule Updates
+
+Stylistic rules are frozen according to [our policy](https://eslint.org/blog/2020/05/changes-to-rules-policies) on how we evaluate new rules and rule changes.
+This means:
+
+* **Bug fixes**: We will still fix bugs in stylistic rules.
+* **New ECMAScript features**: We will also make sure stylistic rules are compatible with new ECMAScript features
```
Comment (mdjermanovic) [inline README.md:196]:
```suggestion
* **New ECMAScript features**: We will also make sure stylistic rules are compatible with new ECMAScript features.
```

**Review by mdjermanovic — APPROVED** [review]
Looks good to me! Added just a missing `.`

**Comment by mdjermanovic** [comment]
Should we also update the Table of Contents?

**Comment by btmills** [comment]
> Should we also update the Table of Contents?

Thanks, added.

**Review by mdjermanovic — APPROVED** [review]

**Pull request merged**